### ISSUE TITLE
coq build fails on netbsd: gmake issue

### ISSUE THREAD

**Issue by azarens** [ocaml/opam-repository#3895]
Note.  It was [suggested](https://github.com/ocaml/opam/issues/2104) that this issue be put here instead of [there](https://github.com/ocaml/opam/issues/).

The configure script fails to run when coq 8.4.5 is being built on a NetBSD box.  The issue lies in that the opam sets the build build process such that an attempt to invoke a make command as `make` is made, even when the packages expects a GNU Make.  The fix is easy: just add `-makecmd gmake` option to the configure script and also replace invocations of `make` with `gmake` when on BSD.

Thus, we should have in the recipe:
```
build: [
  ["./configure" "-makecmd" "%{make}%" "-configdir" "%{lib}%/coq/config" "-mandir" man "-docdir" doc "--prefix" prefix "--usecamlp5" \
"--camlp5dir" "%{lib}%/camlp5" "--coqide" "no"]
  ["%{make}%" "-j%{jobs}%" "world"]
  ["%{make}%" "-j%{jobs}%" "states"]
  ["%{make}%" "install"]
]
```

Note: (in response to [the comment](https://github.com/ocaml/opam/issues/2104#issuecomment-90315946)) merely having `[make "-j%{jobs}%" "world"]` instead of `["%{make}%" "-j%{jobs}%" "world"]` will not work, because `make` would be invoked instead of `gmake` otherwise, and the build would thus break.

The build succeded on my box once this adjustment was introduced.

Please adjust the recipe accordingly.

**Comment by avsm** [comment]
This looks good to me.  A pull request with the patch included would be appreciated.

**Comment by yallop** [comment]
Fixed (#3929)
